app/client/cli/cmd.go

```diff
@@ -2,40 +2,43 @@ package cli
 
 import (
 	"context"
+	"log"
 
-	"github.com/pokt-network/pocket/runtime/configs"
-	"github.com/pokt-network/pocket/runtime/defaults"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
+
+	"github.com/pokt-network/pocket/app/client/cli/flags"
+	"github.com/pokt-network/pocket/runtime/configs"
+	"github.com/pokt-network/pocket/runtime/defaults"
 )
 
 const (
 	cliExecutableName = "p1"
 )
 
 var (
-	remoteCLIURL   string
-	dataDir        string
-	configPath     string
-	nonInteractive bool
-	verbose        bool
-	cfg            *configs.Config
+	cfg *configs.Config
 )
 
 func init() {
-	rootCmd.PersistentFlags().StringVar(&remoteCLIURL, "remote_cli_url", defaults.DefaultRemoteCLIURL, "takes a remote endpoint in the form of <protocol>://<host> (uses RPC Port)")
-	rootCmd.PersistentFlags().BoolVar(&nonInteractive, "non_interactive", false, "if true skips the interactive prompts wherever possible (useful for scripting & automation)")
+	rootCmd.PersistentFlags().StringVar(&flags.RemoteCLIURL, "remote_cli_url", defaults.DefaultRemoteCLIURL, "takes a remote endpoint in the form of <protocol>://<host>:<port> (uses RPC Port)")
+	// ensure that this flag can be overidden by the respective viper-conventional environment variable (i.e. `POCKET_REMOTE_CLI_URL`)
```

Comment:
Have you checked if it environment variable works? I love that you bind a viper configuration instead of looking up the environment variable!

Reply:
I believe so. The CLI client and cluster manager containers are using `POCKET_REMOTE_CLI_URL` in this branch and both k8s and docker-compose localnets seem to be working as expected. I even used a fresh kind cluster for the k8s localnet.

Reply:
:thinking: hmm... it doesn't seem to be working in the context of `make localnet_shell` though when it seems like it should, perhaps I've missed something. Taking another look :eyes:

Reply:
@okdas thanks for making me double check this. :sweat_smile: 

I did a bit of investigating and I'm noticing that the way `BindPFlag()` works is not quite what I expected. It seems to "bind the respective viper key" in the sense that the final result will be reflected in the value retrieved from viper (as opposed to updating the bound flag variable). 0b22fbd adds a line just after the call to `ParseConfig()` which updates the flag var to this final value. My thinking is that it's better to keep the flag var than to replace all references to it with `viper.GetString("remote_cli_url")` or something similar.